src/v1/destinations/hs/networkHandler.js

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+const { TransformerProxyError } = require('../../../v0/util/errorTypes');
+const { prepareProxyRequest, proxyRequest } = require('../../../adapters/network');
+const { isHttpStatusSuccess, getAuthErrCategoryFromStCode } = require('../../../v0/util/index');
+
+const {
+  processAxiosResponse,
+  getDynamicErrorType,
+} = require('../../../adapters/utils/networkUtils');
+const tags = require('../../../v0/util/tags');
+
+const verify = (results, rudderJobMetadata, destinationConfig) => {
+  if (destinationConfig?.apiVersion === undefined) {
+    return false;
+  }
+  if (destinationConfig?.apiVersion === 'newApi') {
+    return Array.isArray(results) && results.length === rudderJobMetadata.length;
+  }
+  return true;
+};
+
+const populateResponseWithDontBatch = (rudderJobMetadata, status, response) => {
+  const errorMessage = JSON.stringify(response);
+  const responseWithIndividualEvents = [];
+  for (const metadata of rudderJobMetadata) {
+    metadata.dontBatch = true;
+    responseWithIndividualEvents.push({
+      statusCode: 500,
+      metadata,
+      error: errorMessage,
+    });
+  }
+  return responseWithIndividualEvents;
+};
+const responseHandler = (responseParams) => {
+  const { destinationResponse, rudderJobMetadata, destinationRequest } = responseParams;
+  const successMessage = `[HUBSPOT Response V1 Handler] - Request Processed Successfully`;
+  const failureMessage =
+    'HUBSPOT: Error in transformer proxy v1 during HUBSPOT response transformation';
+  const responseWithIndividualEvents = [];
+  const { response, status } = destinationResponse;
+
+  if (isHttpStatusSuccess(status)) {
+    // populate different response for each event
+    const results = response?.results;
+    if (verify(results, rudderJobMetadata, destinationRequest?.destinationConfig)) {
+      for (const metadata of rudderJobMetadata) {
+        const proxyOutputObj = {
+          statusCode: 200,
+          metadata,
+          error: 'success',
+        };
+        responseWithIndividualEvents.push(proxyOutputObj);
+      }
+      return {
+        status,
+        message: successMessage,
+        destinationResponse,
```

Comment:
No need of sending `destinationResponse` in case of success server will ignore this. Server uses `response` for updating jobs